go/libraries/doltcore/mvdata/data_mover.go

```diff
@@ -90,15 +90,26 @@ func (dmce *DataMoverCreationError) String() string {
 	return string(dmce.ErrType) + ": " + dmce.Cause.Error()
 }
 
+type GCTableWriteCloser interface {
+	table.TableWriteCloser
+	GC(ctx context.Context) error
+}
+
 // Move is the method that executes the pipeline which will move data from the pipeline's source DataLocation to it's
 // dest DataLocation.  It returns the number of bad rows encountered during import, and an error.
 func (imp *DataMover) Move(ctx context.Context) (badRowCount int64, err error) {
 	defer imp.Rd.Close(ctx)
 	defer func() {
 		closeErr := imp.Wr.Close(ctx)
-		if closeErr != nil {
+		if err == nil {
 			err = closeErr
```

Comment:
I don't like overwriting an existing error if an error closing occurs.

Reply:
Yeah first error is probably the most important